runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryOverlay.java

```diff
@@ -0,0 +1,218 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kourendlibrary;
+
+import com.google.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+
+import java.awt.Polygon;
+import java.awt.geom.Rectangle2D;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Player;
+import net.runelite.api.Point;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+import static net.runelite.api.Perspective.getCanvasTilePoly;
+
+public class KourendLibraryOverlay extends Overlay
+{
+	private final static Point LIBRARY_CENTER = new Point(1632, 3807);
+	private final static int MAXIMUM_DISTANCE = 24;
+
+	private final Library library;
+	private final Client client;
+
+	@Inject
+	KourendLibraryOverlay(Library library, Client client)
+	{
+		this.library = library;
+		this.client = client;
+
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+	}
+
+	@Override
+	public Dimension render(Graphics2D g, java.awt.Point parent)
+	{
+		Player player = client.getLocalPlayer();
+		if (player == null)
+		{
+			return null;
+		}
+
+		Point playerLoc = player.getWorldLocation();
+
+		if (playerLoc.distanceTo(LIBRARY_CENTER) > 45)
+		{
+			return null;
+		}
+
+		List<Bookcase> allBookcases = library.getBookcasesOnLevel(client.getPlane());
+
+		for (Bookcase bookcase : allBookcases)
+		{
+			// AABB
+			Point3D caseLoc = bookcase.getLocation();
+			if (Math.abs(playerLoc.getX() - caseLoc.getX()) > MAXIMUM_DISTANCE
+				|| Math.abs(playerLoc.getY() - caseLoc.getY()) > MAXIMUM_DISTANCE)
+			{
+				continue;
+			}
+
+			Point localBookcase = Perspective.worldToLocal(client, bookcase.getLocation().toPoint());
```

Comment:
You stored bookcase.getLocation in caseLoc already, no need to call that method again here.